src/core/lib/surface/call.c

```diff
@@ -1087,6 +1089,24 @@ static void receiving_initial_metadata_ready(grpc_exec_ctx *exec_ctx,
         &call->metadata_batch[1 /* is_receiving */][0 /* is_trailing */];
     grpc_metadata_batch_filter(md, recv_initial_filter, call);
 
+    /* make sure the received grpc-encoding is amongst the ones listed in
+     * grpc-accept-encoding */
+
+    GPR_ASSERT(call->encodings_accepted_by_peer != 0);
```

Comment:
Sorry for the late response.  

I hope I did not misunderstand the code. As per my understanding, the code at line 265 is client side and the code here is server side..is that correct? - if so, there is a possibility that the client is running an older version of the code (where it doesn't set that LSB) and the server might be running the latest code.  In that case, I am not sure if an assert() on the server side is a good idea.